web/blueprints/facilities/__init__.py

```diff
@@ -613,10 +613,18 @@ def room_patchpanel_json(room_id: int) -> ResponseReturnValue:
 @bp.route("/room/<int:room_id>/tenancies/json")
 def room_tenancies_json(room_id: int) -> ResponseReturnValue:
     room = get_room_or_404(room_id)
+
+    def tenancy_person_id(tenancy) -> int | None:
+        return tenancy.person_id if tenancy.person_id is not None else None
+
+    def tenancy_user_button(tenancy) -> BtnColResponse | None:
+        return user_button(tenancy.user) if tenancy.user is not None else None
+
```

Comment:
Using nested functions with _one_ invocation is nonsense unless you have a really good reason. Here, you could've used a plain variable if you don't like the line length.

Otherwise, if we want to do null coalescence anyway, apart from fancy solutions we could just bake the null coalescence into `user_button` (changing its signature to `User | None -> str | None`)

Reply:
I'm not sure a variable would be applicable here since it is used in a list comprehension. Avoiding long lines was the intention but let's inline these functions if there is not a better way to do it.

Reply:
Right, sorry, I overlooked the list comprehension. Then I'm not sure and I would just vouch for inlining, purely because weakening `user_button` changes semantics.

Reply:
I mean the other way would be the fancy bs way:
```python
def lift[A, B](f: Callable[[A], B | None]) -> Callable[[A | None], B | None]:
    def _lifted(a: A | None) -> B | None:
        return f(a) if a is not None else None
    return _lifted
```
and then use `lift(user_button)(tenancy.user)`

Reply:
(beware, untested)